New PS cmdlet and updates to existing (#9243)
MicrosoftDocs/office-docs-powershell: teams/teams-ps/teams/Get-CsPhoneNumberAssignment.md

```
This example returns information about all activated phone numbers with the capability VoiceApplicationAssignment that are not assigned.

## PARAMETERS

### -ActivationState
Filters the returned results based on the number type. Supported values are Activated, AssignmentPending, AssignmentFailed, UpdatePending and UpdateFailed.

```yaml
Type: System.String
Parameter Sets: (All)
Aliases: 
Applicable: Microsoft Teams

Required: False
Default value: None
Accept pipeline input: False
Accept wildcard characters: False
```

### -AssignedPstnTargetId
Filters the returned results based on the user or resource account id the phone number is assigned to. Supported values are UserPrincipalName, SIP address and ObjectId.

```yaml
Type: System.String
Parameter Sets: (All)
Aliases: 
Applicable: Microsoft Teams

Required: False
Default value: None
Accept pipeline input: False
Accept wildcard characters: False
```

### -CapabilitiesContain
Filters the returned results based on the capabilities assigned to the phone number. You can specify one or more capabilities delimited by comma. Supported capabilities
are ConferenceAssignment, VoiceApplicationAssignment, UserAssignment and OCMobile.

> If you specify only one capability, you will get all phone numbers returned that have that capability assigned. If you specify a comma seperated list for instance like 

Reviewer: ```suggestion
If you specify only one capability, you will get all phone numbers returned that have that capability assigned. If you specify a comma separated list for instance like 

```

---
external help file: Microsoft.Open.Teams.CommonLibrary.dll-Help.xml
Module Name: MicrosoftTeams
online version: https://docs.microsoft.com/powershell/module/teams/get-csphonenumberassignment
applicable: Microsoft Teams
author: jenstrier
ms.author: jenstr
ms.reviewer: 
manager:
schema: 2.0.0
---

# Get-CsPhoneNumberAssignment

## SYNOPSIS

> This cmdlet will display information about one or more phone numbers.

Reviewer 2: ```suggestion
This cmdlet displays information about one or more phone numbers.

```

### Example 3
```powershell
Get-CsPhoneNumberAssignment -CapabilitiesContain "VoiceApplicationAssignment,ConferenceAssignment"
```
This example returns all phone numbers that have both the capability VoiceApplicationAssignment and the capability ConferenceAssignment assigned, but phone numbers that have
only one of these capabilities assigned won't be returned.

### Example 4
```powershell
Get-CsPhoneNumberAssignment -AssignedPstnTargetId user1@contoso.com
```
This example returns information about the phone number assigned to user1@contoso.com.

### Example 5
```powershell
Get-CsPhoneNumberAssignment -AssignedPstnTargetId aa1@contoso.com
```
This example returns information about the phone number assigned to resource account aa1@contoso.com.

### Example 6
```powershell
Get-CsPhoneNumberAssignment -ActivationState Activated -CapabilitiesContain VoiceApplicationAssignment -PstnAssignmentStatus Unassigned
```
This example returns information about all activated phone numbers with the capability VoiceApplicationAssignment that are not assigned.

## PARAMETERS

### -ActivationState

> Filters the returned results based on the number type. Supported values are Activated, AssignmentPending, AssignmentFailed, UpdatePending and UpdateFailed.

Reviewer 2: ```suggestion
Filters the returned results based on the number type. Supported values are Activated, AssignmentPending, AssignmentFailed, UpdatePending, and UpdateFailed.

```

```yaml
Type: System.String
Parameter Sets: (All)
Aliases: 
Applicable: Microsoft Teams

Required: False
Default value: None
Accept pipeline input: False
Accept wildcard characters: False
```

### -AssignedPstnTargetId

> Filters the returned results based on the user or resource account id the phone number is assigned to. Supported values are UserPrincipalName, SIP address and ObjectId.

Reviewer 2: ```suggestion
Filters the returned results based on the user or resource account ID the phone number is assigned to. Supported values are UserPrincipalName, SIP address, and ObjectId.

```

```yaml
Type: System.String
Parameter Sets: (All)
Aliases: 
Applicable: Microsoft Teams

Required: False
Default value: None
Accept pipeline input: False
Accept wildcard characters: False
```

### -CapabilitiesContain

> Filters the returned results based on the capabilities assigned to the phone number. You can specify one or more capabilities delimited by comma. Supported capabilities

Reviewer 2: ```suggestion
Filters the returned results based on the capabilities assigned to the phone number. You can specify one or more capabilities delimited by a comma. Supported capabilities

```

> are ConferenceAssignment, VoiceApplicationAssignment, UserAssignment and OCMobile.

Reviewer 2: ```suggestion
are ConferenceAssignment, VoiceApplicationAssignment, UserAssignment, and OCMobile.

```

If you specify only one capability, you will get all phone numbers returned that have that capability assigned. If you specify a comma separated list for instance like 

> 

Reviewer 2: ```suggestion
```

```

### -CivicAddressId
Filters the returned results based on the CivicAddressId assigned to the phone number. You can get the CivicAddressId by using [Get-CsOnlineLisCivicAddress](/powershell/module/skype/get-csonlineliscivicaddress).

```yaml
Type: System.String
Parameter Sets: (All)
Aliases: 
Applicable: Microsoft Teams

Required: False
Default value: None
Accept pipeline input: False
Accept wildcard characters: False
```

### -IsoCountryCode
Filters the returned results based on the ISO 3166-1 Alpha-2 contry code assigned to the phone number.

```yaml
Type: System.String
Parameter Sets: (All)
Aliases: 
Applicable: Microsoft Teams

Required: False
Default value: None
Accept pipeline input: False
Accept wildcard characters: False
```

### -LocationId
Filters the returned results based on the LocationId assigned to the phone number. You can get the LocationId by using [Get-CsOnlineLisLocation](/powershell/module/skype/get-csonlinelislocation).

```yaml
Type: System.String
Parameter Sets: (All)
Aliases: 
Applicable: Microsoft Teams

Required: False
Default value: None
Accept pipeline input: False
Accept wildcard characters: False
```

### -NumberType

> Filters the returned results based on the number type. Supported values are DirectRouting, CallingPlan and OperatorConnect.

Reviewer 2: ```suggestion
Filters the returned results based on the number type. Supported values are DirectRouting, CallingPlan, and OperatorConnect.

```

```yaml
Type: System.String
Parameter Sets: (All)
Aliases: 
Applicable: Microsoft Teams

Required: False
Default value: None
Accept pipeline input: False
Accept wildcard characters: False
```

### -PstnAssignmentStatus

> Filters the returned results based on the assignment status. Support values are Unassigned, UserAssigned, ConferenceAssigned, VoiceApplicationAssigned and ThirdPartyAppAssigned

Reviewer 2: ```suggestion
Filters the returned results based on the assignment status. Support values are Unassigned, UserAssigned, ConferenceAssigned, VoiceApplicationAssigned, and ThirdPartyAppAssigned.

```

```yaml
Type: System.String
Parameter Sets: (All)
Aliases: 
Applicable: Microsoft Teams

Required: False
Default value: None
Accept pipeline input: False
Accept wildcard characters: False
```

### -Skip

> Skipping the first X returned results and the default value is 0.

Reviewer 2: ```suggestion
Skips the first X returned results and the default value is 0.

```

Default value: None
Accept pipeline input: False
Accept wildcard characters: False
```

### -TelephoneNumberStartsWith
Filters the returned results based on starts with string match for the specified string on TelephoneNumber. For supported formats see TelephoneNumber.

```yaml
Type: System.String
Parameter Sets: (All)
Aliases: 
Applicable: Microsoft Teams

Required: False
Default value: None
Accept pipeline input: False
Accept wildcard characters: False
```

### -Top
Returns the first X returned results and the default value is 500.

```yaml
Type: System.Int32
Parameter Sets: (All)
Aliases: 
Applicable: Microsoft Teams

Required: False
Default value: None
Accept pipeline input: False
Accept wildcard characters: False
```

### CommonParameters
This cmdlet supports the common parameters: -Debug, -ErrorAction, -ErrorVariable, -InformationAction, -InformationVariable, -OutVariable, -OutBuffer, -PipelineVariable, -Verbose, -WarningAction, and -WarningVariable. For more information, see [about_CommonParameters](https://go.microsoft.com/fwlink/?LinkID=113216).

## INPUTS

### None

## OUTPUTS

### ActivationState
The activation state of the telephone number.

### AssignedPstnTargetId

> The Id of the object the phone number is assigned to.

Reviewer 2: ```suggestion
The ID of the object the phone number is assigned to.

```

### Capability
The list of capabilities assigned to the phone number.

### CivicAddressId

> The Id of the CivicAddress assigned to the phone number.

Reviewer 2: ```suggestion
The ID of the CivicAddress assigned to the phone number.

```

### IsoCountryCode
The ISO country code assigned to the phone number.

### LocationId

> The Id of the Location assigned to the phone number.

Reviewer 2: ```suggestion
The ID of the Location assigned to the phone number.

```

### LocationUpdateSupported
Boolean stating if updating of the location assigned to the phone number is allowed.

### NumberType
The type of the phone number.

### PortInOrderStatus
The status of any port in order covering the phone number.

### PstnAssignmentStatus
The assignment status of the phone number.

### PstnPartnerId

> The Id of the PSTN partner providing the phone number.

Reviewer 2: ```suggestion
The ID of the PSTN partner providing the phone number.

```

### PstnPartnerName

> The name of the PSTN partner

Reviewer 2: ```suggestion
The name of the PSTN partner.

```

### TelephoneNumber
The phone number. The number is always displayed with prefixed "+", even if it was not assigned using prefixed "+".

The object returned is of type SkypeTelephoneNumberMgmtCmdletAcquiredTelephoneNumber.

## NOTES
The cmdlet is available in Teams PowerShell module 4.0.0 or later.

> 

Reviewer 2: ```suggestion
```
